lib/google/gax/api_callable.rb

```diff
@@ -33,45 +33,54 @@
 
 module Google
   module Gax
-    # Converts an rpc call into an API call governed by the settings.
+    # Converts an RPC call into an API call.
     #
-    # In typical usage, +func+ will be a proc used to make an rpc request.
-    # This will mostly likely be a bound method from a request stub used to make
-    # an rpc call.
+    # In typical usage, +func+ will be a proc used to make an RPC request.
+    # This will mostly likely be a bound method from a request Stub used to make
+    # an RPC call.
     #
     # The result is created by applying a series of function decorators
-    # defined in this module to +func+.  +settings+ is used to determine
-    # which function decorators to apply.
+    # defined in this module to +func+.
     #
-    # The result is another proc which for most values of +settings+ has the
-    # same signature as the original.
+    # The result is another proc which has the same signature as the original.
     #
     # @param func [Proc] used to make a bare rpc call
-    # @param settings [CallSettings] provides the settings for this call
+    # @param timeout [Numeric] client-side timeout for API calls
+    # @param metadata [Hash]  request headers
     # @param params_extractor [Proc] extracts routing header params from the
```

Comment:
Just a thought (and I'll leave it up to you to decide)... since we're changing a bunch of parameters on this method already should we consider generalizing this one? 

When I read this as-is it's not very clear what this parameter is for and what a routing header is. One option might be to change this to a more generic proc that takes the request and returns a hash to merge with the request headers. This would also prevent us from embedding `x-goog-request-params` by default and we could move that reference into the generator.

Reply:
Or, we might just remove it entirely since you've pulled `metadata` into the parameter list.

Reply:
I am actually leaning towards having more of these procs that do interesting things to the request. We will add another proc for streaming responses, and I would like to wrap responses in `Google::Gax::Operation` or `PagedEnumerable` by providing a proc that does this, rather than by doing in in the client method.

Reply:
Ah, a proc for transforming the output does sounds useful. Let's do that in a new change then and address them all.

Reply:
Yes. I have many, many branches with different changes ready to be posted in new PRs. :)